system-tests/e2e-transfer-test/runner/src/test/java/org/eclipse/dataspaceconnector/test/e2e/AbstractEndToEndTransfer.java

```diff
@@ -50,29 +52,63 @@ void httpPullDataTransfer() {
 
         assertThat(contractAgreementId).isNotEmpty();
 
-        var transferProcessId = CONSUMER.dataRequest(contractAgreementId, assetId, PROVIDER, sync());
+        var dataRequestId = UUID.randomUUID().toString();
+        var transferProcessId = CONSUMER.dataRequest(dataRequestId, contractAgreementId, assetId, PROVIDER, sync());
 
         await().atMost(timeout).untilAsserted(() -> {
             var state = CONSUMER.getTransferProcessState(transferProcessId);
             assertThat(state).isEqualTo(COMPLETED.name());
         });
 
-        await().atMost(timeout).untilAsserted(() -> {
-            given()
-                    .baseUri(CONSUMER.backendService().toString())
-                    .when()
-                    .get("/api/consumer/data")
-                    .then()
-                    .statusCode(200)
-                    .body("message", equalTo("some information"));
-        });
+        // retrieve the data reference
+        await().atMost(timeout).untilAsserted(() -> given()
```

Comment:
As the `EndpointDataReference` object is used below, this could be extacted in a method and moved in the `Participant` class, like it has been done for the `getContractAgreementId` method, this will make the tests more readable and it will encapsulate the url of the call.